main.tf

```diff
@@ -1,50 +1,56 @@
 module "s3" {
   source     = "./modules/s3"
-  model_name = var.model_name
+  model_name = local.model_name
   tags       = var.tags
 }
 
 
 module "sagemaker" {
   source                       = "./modules/sagemaker"
   tags                         = var.tags
-  sagemaker_execution_role_arn = module.iam.sagemaker_role_arn
+  sagemaker_execution_role_arn = module.iam.sagemaker_role.arn
   model_target_variable        = var.model_target_variable
   algorithm_choice             = var.algorithm_choice
   tuning_metric                = var.tuning_metric
-  config_bucket_id             = module.s3.config_bucket_id
+  config_bucket_id             = module.s3.config_bucket.id
+  data_s3_bucket               = var.data_s3_bucket
```

Comment:
Passing in the bucket name and filepath separately allows us to target the bucket independently in IAM policies